### ISSUE TITLE
Save the data from the trigger function (ex. data from API call)

### ISSUE THREAD

**Issue by jejopl** [PresearchOfficial/presearch-packages#31]
**Is your feature request related to a problem? Please describe.**
Example, weather app: On the `trigger()` function I want to fetch data from the weather API provider just to check if they have some data for the query. And if they do, I want to trigger the package. But the problem is I need to call the API again in the actual `package function` to get the necessary data. 

**Describe the solution you'd like**
Refactor `trigger()` function. I imagine that currently, the server is checking if `<package> trigger() == true` and if is, it's executing `<package> function()`. I don't know if this will be a good solution but the `trigger()` function can return an array or object, where the first item will be a boolean, and the second item can be a custom value. And then you will be able to read this value from the `<package> function()` - in this case, the data from the API

**Describe alternatives you've considered**

**Additional context**
I'm still learning JS and I don't know if I can achieve this already. I know that I can use a global variable for this but I don't think that this will be the right way to do it

**Comment by davidkeefe24** [comment]
This is in the works for a future release.

**Comment by davidkeefe24** [comment]
I have implemented this locally with a pre-condition that needs to be met before it makes the API call. I have also modified the trigger function handler to accept data as a return and pass it to the package function as a new parameter. 

I feel this approach can solve many use cases where a specific word would be used in the query but also allow to query an API for qualifying data, like weather or map triggers. We wouldn't have to have the crazy long list of cities or matching queries.

For example, a pre-condition could be the query must include word "weather", if it does then you hit the weather API with the rest of the details in the query string. This can get a little tricky because we wouldn't want to pass "weather Toronto" to the API, only "Toronto", so handling cases like that are going to arise kind of on a package by package basis.

@jejopl I'm curious about your thoughts and if it would help meet your requirements?
